File: cache-caffeine/src/test/java/io/micronaut/cache/ConditionalService.java
package io.micronaut.cache;

import io.micronaut.cache.annotation.CacheConfig;
import io.micronaut.cache.annotation.Cacheable;
import io.micronaut.context.annotation.Requires;
import jakarta.inject.Singleton;

@Singleton
@CacheConfig(cacheNames = {"conditional"})
@Requires(property = "spec.name", value = "ConditionalCacheSpec")
public class ConditionalService {

    DocRepo repsoitory = new DocRepo();

    // tag::conditional[]
    @Cacheable(condition = "#{id > 5}")
    public String get(Integer id) {
        return repsoitory.get(id);
    }
    // end::conditional[]

    // here to make the docs look nice when we extract the function above
    private static class DocRepo {

        String get(Integer id) {
            return "test " + id + " " + System.currentTimeMillis();
        }
    }
}

File: test-suite-caffeine-python/src/test/python/micronaut/cache/docs/ConditionalService.py
import time
from dataclasses import dataclass

from jakarta.inject import Singleton
from micronaut.cache.annotation import CacheConfig, Cacheable
from micronaut.core.annotation import Introspected


# tag::conditional[]
@Introspected
@dataclass
class Id:
    value: int
# end::conditional[]


# here to make the docs look nice when we extract the function below
class DocRepo:

    def get(self, id: Id) -> str:
        return f"test {id.value} {time.time_ns() // 1_000_000}"


@Singleton
@CacheConfig(cacheNames=["conditional"])
class ConditionalService:

    def __init__(self):
        self.repository = DocRepo()
    # tag::conditional[]

    @Cacheable(condition="#{id.value > 5}")
    def get(self, id: Id) -> str:
        return self.repository.get(id)
    # end::conditional[]
